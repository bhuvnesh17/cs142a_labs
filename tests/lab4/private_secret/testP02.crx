// test operator precedence

func main() : void {
  var a:int;
  var b:int;
  var c:int;
  var d:int;

  let d = a + b / c;
  let d = a - b / c;
  let d = a + b * c;
  let d = a - b * c;
  let d = a / b + c;
  let d = a * b + c;
  let d = a / b - c;
  let d = a * b - c;
}
