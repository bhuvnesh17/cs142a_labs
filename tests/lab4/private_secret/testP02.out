ast.DeclarationList(3,1)
  ast.FunctionDefinition(3,1)[Symbol(main), []]
    ast.StatementList(4,3)
      ast.VariableDeclaration(4,3)[Symbol(a)]
      ast.VariableDeclaration(5,3)[Symbol(b)]
      ast.VariableDeclaration(6,3)[Symbol(c)]
      ast.VariableDeclaration(7,3)[Symbol(d)]
      ast.Assignment(9,3)
        ast.AddressOf(9,7)[d]
        ast.Addition(9,13)
          ast.Dereference(9,11)
            ast.AddressOf(9,11)[a]
          ast.Division(9,17)
            ast.Dereference(9,15)
              ast.AddressOf(9,15)[b]
            ast.Dereference(9,19)
              ast.AddressOf(9,19)[c]
      ast.Assignment(10,3)
        ast.AddressOf(10,7)[d]
        ast.Subtraction(10,13)
          ast.Dereference(10,11)
            ast.AddressOf(10,11)[a]
          ast.Division(10,17)
            ast.Dereference(10,15)
              ast.AddressOf(10,15)[b]
            ast.Dereference(10,19)
              ast.AddressOf(10,19)[c]
      ast.Assignment(11,3)
        ast.AddressOf(11,7)[d]
        ast.Addition(11,13)
          ast.Dereference(11,11)
            ast.AddressOf(11,11)[a]
          ast.Multiplication(11,17)
            ast.Dereference(11,15)
              ast.AddressOf(11,15)[b]
            ast.Dereference(11,19)
              ast.AddressOf(11,19)[c]
      ast.Assignment(12,3)
        ast.AddressOf(12,7)[d]
        ast.Subtraction(12,13)
          ast.Dereference(12,11)
            ast.AddressOf(12,11)[a]
          ast.Multiplication(12,17)
            ast.Dereference(12,15)
              ast.AddressOf(12,15)[b]
            ast.Dereference(12,19)
              ast.AddressOf(12,19)[c]
      ast.Assignment(13,3)
        ast.AddressOf(13,7)[d]
        ast.Addition(13,17)
          ast.Division(13,13)
            ast.Dereference(13,11)
              ast.AddressOf(13,11)[a]
            ast.Dereference(13,15)
              ast.AddressOf(13,15)[b]
          ast.Dereference(13,19)
            ast.AddressOf(13,19)[c]
      ast.Assignment(14,3)
        ast.AddressOf(14,7)[d]
        ast.Addition(14,17)
          ast.Multiplication(14,13)
            ast.Dereference(14,11)
              ast.AddressOf(14,11)[a]
            ast.Dereference(14,15)
              ast.AddressOf(14,15)[b]
          ast.Dereference(14,19)
            ast.AddressOf(14,19)[c]
      ast.Assignment(15,3)
        ast.AddressOf(15,7)[d]
        ast.Subtraction(15,17)
          ast.Division(15,13)
            ast.Dereference(15,11)
              ast.AddressOf(15,11)[a]
            ast.Dereference(15,15)
              ast.AddressOf(15,15)[b]
          ast.Dereference(15,19)
            ast.AddressOf(15,19)[c]
      ast.Assignment(16,3)
        ast.AddressOf(16,7)[d]
        ast.Subtraction(16,17)
          ast.Multiplication(16,13)
            ast.Dereference(16,11)
              ast.AddressOf(16,11)[a]
            ast.Dereference(16,15)
              ast.AddressOf(16,15)[b]
          ast.Dereference(16,19)
            ast.AddressOf(16,19)[c]

